core/perf_test/PerfTest_Stream.cpp

```diff
@@ -0,0 +1,284 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/**
+ * @file PerfTest_Stream.cpp
+ * @brief Implementation of STREAM benchmark operations for Kokkos.
+ *
+ * @details This file provides a set of memory bandwidth benchmarks based on the
+ * STREAM benchmark suite. It implements the five core STREAM operations (Set,
+ * Copy, Scale, Add, and Triad) using Kokkos parallel primitives. It includes
+ * validation.
+ *
+ * The implementation strives to use as few Kokkos features as possible, thus
+ * validation is performed on the host rather than via parallel_reduce.
+ */
+
+#include <Kokkos_Core.hpp>
+#include <benchmark/benchmark.h>
+#include "Benchmark_Context.hpp"
+
+constexpr static double A_INIT = 1.0;
+constexpr static double B_INIT = 2.0;
+constexpr static double C_INIT = 3.0;
+constexpr static double SCALAR = 4.0;
+
+template <unsigned MemTraits>
+using StreamView = Kokkos::View<double*, Kokkos::MemoryTraits<MemTraits>>;
+
+// different than benchmarks/stream, which uses int
+// wide index types are common as GPU memory grows
+using StreamIndex = int64_t;
+using Policy      = Kokkos::RangePolicy<Kokkos::IndexType<StreamIndex>>;
+
+template <typename V>
+void perform_set(const V& a, const double scalar) {
+  Kokkos::parallel_for(
+      "set", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = scalar; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_copy(const V& a, const V& b) {
+  Kokkos::parallel_for(
+      "copy", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = a[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_scale(const V& b, const V& c, const double scalar) {
+  Kokkos::parallel_for(
+      "scale", Policy(0, b.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = scalar * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_add(const V& a, const V& b, const V& c) {
+  Kokkos::parallel_for(
+      "add", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { c[i] = a[i] + b[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_triad(const V& a, const V& b, const V& c, const double scalar) {
+  Kokkos::parallel_for(
+      "triad", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = b[i] + scalar * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+int validate_array(V& a_dev, const double expected) {
+  const auto a = Kokkos::create_mirror_view(Kokkos::WithoutInitializing, a_dev);
+  Kokkos::deep_copy(a, a_dev);
+
+  double error = 0.0;
+  for (size_t i = 0; i < a.size(); ++i) {
+    error += std::abs(a[i] - expected);
+  }
+  const double avgError = error / (double)a.size();
+
+  constexpr double epsilon = 1.0e-13;
+  return std::abs(avgError / expected) > epsilon;
+}
+
+template <unsigned MemTraits>
+static void StreamSet(benchmark::State& state) {
+  const size_t N8                 = std::pow(state.range(0), 8);
```

Comment:
Why are we not just passing the extent (10^8) as argument?

Reply:
This mirrors how the view benchmarks do things. Happy to change it how you suggest if you prefer.